### PR TITLE
#321 - Fixes Master Password bypasses.

### PR DESCRIPTION
**Pull request by NiklasSchmitt**
Fixes issues #320 & #321

### CODE REVIEW

File: html/browser_action/views/password_prompt.html [hunk 1]
```diff
@@ -4,8 +4,8 @@
             <img src="/icons/passman_logo.png" height="80"/>
             <div>{{'extension_locked'| translate}}:</div>
             <div class="group">
-                <input autofocus type="password" ng-enter="apply_settings()" placeholder="{{'password' | translate}}"
-                       ng-model="master_password">
+                <input autofocus required type="password" ng-enter="apply_settings()" placeholder="{{'password' | translate}}"
+                       id="master_password" ng-model="master_password">
```
Comment (binsky08) [inline html/browser_action/views/password_prompt.html:8]: why do you give it an id? is it required by anything?

File: js/ui/popup/controllers/setup.js [hunk 2]
```diff
@@ -93,11 +93,11 @@
                     });
                 },
                 vault: function (callback) {
-                    try {
-                        PAPI.decryptString($scope.settings.default_vault.challenge_password, $scope.settings.vault_password);
+                    var decryptedString = PAPI.decryptString($scope.settings.default_vault.challenge_password, $scope.settings.vault_password);
+
+                    if (decryptedString && decryptedString !== false) {
```
Comment (binsky08) [inline js/ui/popup/controllers/setup.js:98]: If the first assumption of the condition is false, the condition fails either

So a simple `if (PAPI.decryptString($scope.settings.default_vault.challenge_password, $scope.settings.vault_password) !== false)` would clarify the condition.

File: js/lib/api.js [hunk 3]
```diff
@@ -28,15 +28,17 @@ window.PAPI = (function () {
...
             ciphertext = window.atob(ciphertext);
             var rp = {};
             try {
                 /** global: sjcl */
                 return sjcl.decrypt(_key, ciphertext, encryption_config, rp);
             } catch (e) {
-                throw e;
+                //throw e;
+                console.warn(e);
+                return false;
```
Comment (binsky08) [inline js/lib/api.js:41]: Rework required.
You have to check all code that is using this decryptString function if you really want to change that behaviour.
There is code that requires on how the error handling with exceptions is currently handled. For example `decryptCredential` in line 44 of this file.

File: html/browser_action/views/password_prompt.html (hunk 1 above)
Comment (NiklasSchmitt) [reply html/browser_action/views/password_prompt.html:8]: No ID required. Will be removed again.

**Comment by NiklasSchmitt** [comment]
I'll close this pull request and will try another solution to fix the mentioned issues.

**Pull request closed without merging**